### PR TITLE
Add a pkg custom events and start using it for storage cluster

### PR DESCRIPTION
**Pull request by iamniting**
The PR contains a new pkg `customevents` which will allow an operator to limit the events and stop spamming the APIServer.

cc @raghavendra-talur @rohantmp 

PS: I have added this as a new pkg because we need this at the rook layer also.

### CODE REVIEW

**Comment by iamniting** [comment]
Here is an example output

```
Events:
  Type     Reason            Age                     From                       Message
  ----     ------            ----                    ----                       -------
  Warning  UninstallPending  5m2s                    controller_storagecluster  Uninstall: Waiting on NooBaa system noobaa to be deleted
  Warning  UninstallPending  4m23s (x10 over 4m58s)  controller_storagecluster  Uninstall: Waiting for cephCluster to be deleted

```

```
Events:
  Type     Reason            Age                 From                       Message
  ----     ------            ----                ----                       -------
  Warning  UninstallPending  5m13s               controller_storagecluster  Uninstall: Waiting on NooBaa system noobaa to be deleted
  Warning  UninstallPending  1s (x13 over 5m9s)  controller_storagecluster  Uninstall: Waiting for cephCluster to be deleted
```

It is generating the same events 10 times within 5 minutes and a new cycle goes on after every 5 minutes. 

PS: time and count are configurable

File: pkg/customevents/events.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	// report events x times where x is count
+	count int
+
+	// report events after x minutes
+	eventReportAfterMinutes int
+}
+
+// NewEventReporter returns EventReporter object
+func NewEventReporter(recorder record.EventRecorder, count, reportAfter int) *EventReporter {
```
Comment (raghavendra-talur) [inline pkg/customevents/events.go:26]: If possible think of expanding `count` to `maxCount` or something similar to make it known to the user that it is the max repetition for a particular event.

**Comment by jarrpa** [comment]
When you say "we need this at the rook layer also", do you mean in the rook/rook repo? If that's the case then I hope you're not planning on importing this into rook/rook from this repo...

**Comment by rohantmp** [comment]
> importing this into rook/rook from this repo

Beat me to it. ocs-operator does not really have versions beyond the release branches, and since we import rook, the versioning will be hell even if we don't run into cyclic dep issues. 

I raised the question of where to put this during the ocs-operator huddle yesterday and since the functionality is relatively small, we can just duplicate it in each repo.

File: pkg/customevents/events.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
+package customevents
```
Comment (rohantmp) [inline pkg/customevents/events.go:1]: eventutil or something else would be more accurate

Comment (rohantmp) [reply pkg/customevents/events.go:1]: Oh, I see that some event code already exists in the util package. Maybe this should just add to that?

Comment (jarrpa) [reply pkg/customevents/events.go:1]: Yeah, probably best to move it into `util/events.go`.

File: pkg/customevents/events.go (hunk 1 above)
Comment (iamniting) [reply pkg/customevents/events.go]: Done

File: pkg/customevents/events.go (hunk 2 above)
Comment (iamniting) [reply pkg/customevents/events.go]: Done

**Comment by iamniting** [comment]
> When you say "we need this at the rook layer also", do you mean in the rook/rook repo? If that's the case then I hope you're not planning on importing this into rook/rook from this repo...

Yes I was planning to import from this repo, But it looks like duplicating code is better option.

**Comment by iamniting** [comment]
@jose Can we take a look at this one

**Review by jarrpa — APPROVED** [review]
Sorry for the delay in review!

Looks good. Offhand I have an instinct to ask for the first and last commits to be squashed, but I think I can live with this.

/lgtm

**Comment by iamniting** [comment]
> Sorry for the delay in review!

No Issues

**Pull request merged**